packages/firestore/src/model/values.ts

```diff
@@ -73,6 +81,10 @@ export function typeOrder(value: Value): TypeOrder {
 
 /** Tests `left` and `right` for equality based on the backend semantics. */
 export function valueEquals(left: Value, right: Value): boolean {
+  if (left === right) {
```

Comment:
I don't fully follow here, were we always able to do this direct comparison, or does this PR now allow us to do this direct comparison?

Reply:
We were always able to do this. I added this now to make it more performant to do an equals check on MAX_VALUE.